File: flutter-app/lib/models/health_metrics_request.dart
class HealthMetricsRequest {
  final String userId;
  final int heightFeet;
  final int heightInches;
  final double weightKg;
  final String goal;

  HealthMetricsRequest({
    required this.userId,
    required this.heightFeet,
    required this.heightInches,
    required this.weightKg,
    required this.goal,
  });

  Map<String, dynamic> toJson() {
    return {
      'user_id': userId,
      'height_feet': heightFeet,
      'height_inches': heightInches,
      'weight_kg': weightKg,
      'goal': goal,
    };
  }
}

File: spring-backend/src/main/java/com/scanmyfood/backend/dto/HealthMetricsRequest.java
package com.scanmyfood.backend.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class HealthMetricsRequest {
    private String firebaseUid;
    private int heightFeet;
    private int heightInches;
    private double weightKg;
    private String goal;
}
